enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,78 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@egarcia"
+  - "@adduarte"
+reviewers:
+  - "@mandre"
+  - "@pierre"
+approvers:
+  - "@mandre"
+  - "@pierre"
+creation-date: 2020-5-26
+last-updated: 2020-5-26
+status: implementable
+---
+
+# OpenStack BYO External Connectivity
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up exeternal connectivity to the OpenShift cluster. Depending on how your cluster is 
+configured, the usage of floating IPs may not be possible or convenient in OpenStack. Our goal is to support configuring external 
+connectivity to the installer without the use of floating IPs.
+
+
+### Non-Goals
+
+This enhancement does not intend to provide any guidance or features that go further than allowing customers to customize entrypoint connectivity. Operations like attaching loadbalancers to the entrypoints, or any day 2 operations are beyond the scope of this enhancement. 
+
+## Proposal
+
+In order to implement this feature fully, the following requirements must be met:
+    1. Allow users pass pre-configured ports for the Ingress and API VIPs
```

Comment:
Do we need the ingress fip as well as api. The current installation expects the user to create the ingress fip themselves by hand: https://docs.openshift.com/container-platform/4.3/installing/installing_openstack/installing-openstack-installer-custom.html#installation-osp-accessing-api-floating_installing-openstack-installer-custom
(https://issues.redhat.com/browse/OSASINFRA-932). The installer doesn't require the ingress fip to be created.  just the load balancer one.

Reply:
VIP not FIP. The solution I am proposing is to let users pass a custom port to the installer for each VIP

Reply:
We shouldn't have to pass pre-configured ports for the Ingress and API VIPs -- we already allow setting the API and Ingress VIP addresses via `apiVIP` and `ingressVIP` params and that should be enough for creating the ports on the node subnet. I'd remove this statement.

Instead explicitly say that the users should set the `apiVIP` to an IP address on the machineNetwork and have a `api.<base_domain>` DNS record that points to it. The node running the installer needs to have L3 connectivity to the subnet otherwise the installation appears to time out (but may still succeed), in consequence it's recommended to not let the installer create the network and use user-provided networks instead.

Reply:
Gotcha. I'll remove this for now and clarify this. Is there a use case where people only want the VIPs ports on the provider network, and the rest of the installer ports on private tenant network?

Reply:
Good question. I don't think so (by design the VIP is picked from the node subnet), but if that's the case I guess this can be done as a day 2 operation.

Reply:
Fair enough. For now, I will leave it out because I don't want to invent use cases. I'll leave the thread just in case it is a real use case so someone can come change our minds :)

Reply:
at least we don't have such use case, either private network with FIP or pure provider network should be good enough